sshmux.go

```diff
@@ -371,11 +384,14 @@ func sshmuxServer(configFile string) {
 		}
 		sshConfig.AddHostKey(key)
 	}
+
 	listener, err := net.Listen("tcp", config.Address)
 	if err != nil {
 		log.Fatal(err)
 	}
+	listener = &proxyproto.Listener{Listener: listener}
```

Comment:
Shouldn't this stay behind a config option to prevent client from spoofing the source address when undesired?

Reply:
🤔 In production there's usually a CIDR allow list from which `PROXY` headers are accepted, which is also recommended here IMO? Thanks for pointing out the security hole.

Now there's a question that when an unauthorized client sends a `PROXY` header shall we reject it (shut down the connection) or simply ignore it?

Reply:
In a production environment you don't expose a PROXY-enabled service to the outside, while you can still launch separate instances with and without PROXY protocol to accept connections from different sources.

Reply:
Well, it's not really possible to spawn two different `sshmux` instances within `sshmux_tests`, without some fundamental changes...

I would still prefer using TCP source IP here. If you want to completely disable PROXY protocol, you can leave the allow list empty; and if you want to have a single upstream proxy, you can allow it explicitly.

Reply:
> spawn two different `sshmux` instances within `sshmux_tests`

There's one way to do that: Move the part that reads and parses the config file out of `sshmuxServer`, then you can start a second server after modifying the `config` object.